tfhe/src/integer/server_key/radix_parallel/slice.rs

```diff
@@ -0,0 +1,362 @@
+use std::ops::RangeBounds;
+
+use rayon::prelude::*;
+
+use crate::error::InvalidRangeError;
+use crate::integer::server_key::radix::slice::{parse_bounds, slice_oneblock_clear_unaligned};
+use crate::integer::{RadixCiphertext, ServerKey};
+use crate::prelude::{CastFrom, CastInto};
+
+impl ServerKey {
+    /// Extract a slice from a ciphertext. The size of the slice is a multiple of the block
+    /// size but it is not aligned on block boundaries, so we need to mix block n and (n+1) to
+    /// create a new block, using the lut function `slice_oneblock_clear_unaligned`.
+    fn scalar_blockslice_unaligned_parallelized(
+        &self,
+        ctxt: &RadixCiphertext,
+        start_block: usize,
+        block_count: usize,
+        offset: usize,
```

Comment:
I'm wondering how much do we want to do pre-emptive checks on these, to make sure the code panics is start, block count would yield invalid indexes later in the parallel code. and also check offset < message_modulus.ilog2

Reply:
done